### PR TITLE
Switch supporter store page to use usercards

### PR DESCRIPTION
**Pull request by notbakaneko**
Change the store supporter page to use usercards and some layout adjustments

### CODE REVIEW

File: resources/assets/coffee/react/_components/friend-button.coffee [hunk 1]
```diff
@@ -137,5 +137,6 @@ class @FriendButton extends React.PureComponent
     # - not viewing own card
     # - already a friend or can add more friends
     currentUser.id? &&
+      !isNaN(@props.user_id) &&
```
Comment (nanaya) [inline resources/assets/coffee/react/_components/friend-button.coffee]: `_.isFinite`?

File: resources/assets/less/bem/store-slider.less [hunk 2]
```diff
@@ -17,6 +17,17 @@
...
+  @preset-font-size: 12px;
+  @preset-height: 20px;
+  @preset-margin: 4px;
+  @preset-width: 36px;
+  @callout-padding: 6px;
+  @callout-extension: 64px;
+
+  // inset slider to prevent overflows making the page wider.
+  // the 2px is to add a gap between the callout and the page edge.
+  padding: 0 @callout-extension / 2 - @callout-padding * 2 + 2px;
```
Comment (nanaya) [inline resources/assets/less/bem/store-slider.less]: here have some brackets `()` ಠ_ಠ

File: resources/assets/less/bem/store-slider.less [hunk 3]
```diff
@@ -62,11 +78,11 @@
     background-color: #666;
     color: #fff;
     cursor: pointer;
-    font-size: 12px;
-    margin: 4px;
-    height: 20px;
-    width: 36px;
-    line-height: 19px;
+    font-size: @preset-font-size;
```
Comment (nanaya) [inline resources/assets/less/bem/store-slider.less]: set in `__presets` instead? (also for the blurb thing)

File: resources/views/store/products/supporter-tag.blade.php [hunk 4]
```diff
@@ -53,7 +60,8 @@
                 </div>
             </div>
         </div>
-        <div class="grid grid--xs grid--packed">
+        <div class="grid grid--xs grid--right store-slider__presets">
+            <span class="store-slider__presets-blurb">months</span>
```
Comment (nanaya) [inline resources/views/store/products/supporter-tag.blade.php]: localization?

File: resources/views/store/products/supporter-tag.blade.php [hunk 5]
```diff
@@ -53,7 +60,8 @@
...
             @foreach([2, 4, 6, 8, 12, 18, 24] as $months)
```
Comment (nanaya) [inline resources/views/store/products/supporter-tag.blade.php]: add 1, remove 8?

File: resources/assets/less/bem/store-slider.less [hunk 6]
```diff
@@ -84,6 +100,15 @@
     }
   }
 
+  &__presets {
+    align-items: center;
```
Comment (nanaya) [inline resources/assets/less/bem/store-slider.less]: where's the rest of flex-- oh ಠ_ಠ

File: resources/assets/less/bem/store-slider.less (hunk 2 above)
Comment (notbakaneko) [reply resources/assets/less/bem/store-slider.less]: but operator precedence :trollface:

File: resources/assets/coffee/react/_components/friend-button.coffee [hunk 7]
```diff
@@ -137,5 +137,6 @@ class @FriendButton extends React.PureComponent
     # - not viewing own card
     # - already a friend or can add more friends
     currentUser.id? &&
+      _.isFinite(@props.user_id) &&
```
Comment (nanaya) [inline resources/assets/coffee/react/_components/friend-button.coffee]: wait why does this not just do `@props.user_id?` just like the line right above it?

Comment (nanaya) [reply resources/assets/coffee/react/_components/friend-button.coffee]: oh wtf it parsed `null` (string) as `NaN` ಠ_ಠ

**Pull request merged**